core/vm/contracts_suave_test.go

```diff
@@ -0,0 +1,132 @@
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+	"github.com/stretchr/testify/require"
+)
+
+type mockSuaveBackend struct {
+}
+
+func (m *mockSuaveBackend) Initialize(bid suave.Bid, key string, value []byte) (suave.Bid, error) {
+	return suave.Bid{}, nil
+}
+
+func (m *mockSuaveBackend) Store(bidId suave.BidId, caller common.Address, key string, value []byte) (suave.Bid, error) {
+	return suave.Bid{}, nil
+}
+
+func (m *mockSuaveBackend) Retrieve(bid suave.BidId, caller common.Address, key string) ([]byte, error) {
+	return nil, nil
+}
+
+func (m *mockSuaveBackend) SubmitBid(suave.Bid) error {
+	return nil
+}
+
+func (m *mockSuaveBackend) FetchBidById(suave.BidId) (suave.Bid, error) {
+	return suave.Bid{}, nil
+}
+
+func (m *mockSuaveBackend) FetchBidsByProtocolAndBlock(blockNumber uint64, namespace string) []suave.Bid {
+	return nil
+}
+
+func (m *mockSuaveBackend) BuildEthBlock(ctx context.Context, args *suave.BuildBlockArgs, txs types.Transactions) (*engine.ExecutionPayloadEnvelope, error) {
+	return nil, nil
+}
+
+func (m *mockSuaveBackend) BuildEthBlockFromBundles(ctx context.Context, args *suave.BuildBlockArgs, bundles []types.SBundle) (*engine.ExecutionPayloadEnvelope, error) {
+	return nil, nil
+}
+
+var dummyBlockContext = BlockContext{
+	CanTransfer: func(StateDB, common.Address, *big.Int) bool { return true },
+	Transfer:    func(StateDB, common.Address, common.Address, *big.Int) {},
+	BlockNumber: big.NewInt(0),
+}
+
+func TestSuavePrecompileStub(t *testing.T) {
+	// This test ensures that the Suave precompile stubs work as expected
+	// for encoding/decoding.
+	mockSuaveBackend := &mockSuaveBackend{}
+	suaveBackend := &SuaveExecutionBackend{
+		ConfiendialStoreBackend: mockSuaveBackend,
+		MempoolBackend:          mockSuaveBackend,
+		OffchainEthBackend:      mockSuaveBackend,
+	}
+
+	statedb, _ := state.New(types.EmptyRootHash, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+	vmenv := NewOffchainEVM(suaveBackend, dummyBlockContext, TxContext{}, statedb, params.AllEthashProtocolChanges, Config{IsOffchain: true})
+
+	methods := map[string]common.Address{
+		"extractHint":               extractHintAddress,
+		"fetchBids":                 fetchBidsAddress,
+		"newBid":                    newBidAddress,
+		"simulateBundle":            simulateBundleAddress,
+		"submitEthBlockBidToRelay":  submitEthBlockBidToRelayAddress,
+		"buildEthBlock":             buildEthBlockAddress,
+		"confidentialStoreRetrieve": confStoreRetrieveAddress,
+		"confidentialStoreStore":    confStoreStoreAddress,
+	}
+
+	// The objective of the unit test is to make sure that the encoding of the precompile
```

Comment:
To highlight this remark. Once code generation is in place. This hack can be removed.